MIGRATION.md

```diff
@@ -0,0 +1,112 @@
+Migrating from Jbuilder to Dune
+===============================
+
+Dune was initially called Jbuilder. Up to mid-2018, the package was
+still called `jbuilder` and it was only installing a `jbuilder`
+binary. This document explain how the migration to Dune will happen.
+
+Timeline
+--------
+
+The general idea is that the migration is gradual and existing
+Jbuilder projects don't need to be updated all at once. We encourage
+users to switch their development repositories and continue their
+usual release cycle. There is no need to re-release existing packages
+just to switch to Dune immediately.
+
+The plan is as follow:
+
+### July 2018: release of Dune 1.0.0
+
+First release of the opam package `dune`. The `jbuilder` package
+becomes a transition package that depends on `dune`.
+
+The `dune` package installs two binaries: `dune` and `jbuilder`. These
+two binaries are exactly the same and they work on both Jbuilder and
+Dune projects. Additionally they recognize both Jbuilder and Dune
+configuration files. The new Dune configuration files are described
+later in this document.
+
+### January 2019: deprecation of Jbuilder
+
+At this point, the `jbuilder` binary emits a warning on every startup
+inviting users to switch to `dune`. When encountering `jbuild` or
+other Jbuilder configuration files, both binaries emit a warning. The
+rest is unchanged.
+
+During this period, it makes sense for projects to do new releases
+just to switch to Dune if none of their existing releases is using
+Dune.
+
+### July 2019: support for Jbuilder is dropped
+
+`jbuilder` is now a dummy executable that always exit with an error
+message on startup. `dune` no longer reads `jbuild` or other Jbuidler
```

Comment:
If it is the `jbuilder` _package_ is also updated to do that, one must make sure all _existing_ packages in opam are updated with an upper bound on `jbuilder` or they all become uninstallable.

Reply:
The `jbuilder` package will be unchanged, but we will indeed need to add a conflict with newer versions of `dune`. I added a comment about it.